pages/api/studentPortal/quizzes/scoringLogicFPL/computeScore.test.ts

```diff
@@ -0,0 +1,76 @@
+import { QuizViewState } from "../../../../../components/resources/quizzes/shared/types";
+import { computeScore } from "./computeScore";
+
+  test("Test computeScore function - return object with scores for each programming language", async () => {
+    //set 
+    //intialized a simplified quizViewState object
+    const quizViewState:QuizViewState = {
+        title:"",
+        body: "",
+        currentQuestion: 1,
+        progress: 1,
+        questions: [
+          {
+            title: "",
+            body: "",
+            options: [
+              {
+                name:"",
+                isSelected: true,
+                result: 'JavaScript',
+                weight: 10,
+              },
+              {
+                name:"",
+                isSelected: false,
+                result: 'HTML/CSS',
+                weight: 5,
+              },
+              {
+                name:"",
+                isSelected: true,
+                result: 'Python',
+                weight: 5,
+              },
+            ],
+          },
+          {
+            title: "",
+            body: "",
+            options: [
+              {
+                name:"",
+                isSelected: false,
+                result: 'JavaScript',
+                weight: 5,
+              },
+              {
+                name:"",
+                isSelected: true,
+                result: 'HTML/CSS',
+                weight: 10,
+              },
+              {
+                name:"",
+                isSelected: true,
+                result: 'Python',
+                weight: 20,
+              },
+            ],
+          },
+        ],
+      };
+
+      const expectedScore = {
+        JavaScript: 10,
+        'HTML/CSS': 10,
+        Python: 25,
+      };
+  
+    // Act
+    const actualScore = computeScore(quizViewState);
+  
+    // Assert
+    expect(actualScore).toStrictEqual(expectedScore);
+  });
+  
```

Comment:
Hey Vithushan - since this function solely summates weight values - aren't some of these tests redundant?? 

I feel that the preferredLanguageQuiz component would benefit from these edge-cases!

What do you think?

Reply:
@jimip85 makes sense, if you feel like the tests are redundant, you don't have to add them. But I'm in the camp of better being safe than sorry. 

I believe that we may need to adapt your scoring logic in the future to do more than just calculate weights, and if the complexity of score calculations grow and change, we'd want to make sure these tests always pass. 

Having more complete coverage will help protect your feature as other people work on it, so I'm generally ok to add more test coverage. If you truly feel that they're pointless then feel free to push back.

I'm good with keeping the tests here